Issue 15637: Add metatype config and constants for Kerberos for Ldap

kristip17 opened the pull request:
Fixes #15637 

Just delivering the metatype and constants in this PR to make sure we make the translation date for the metatype.properties.

skasund commented:
L2 message review not required

kristip17 commented:
#build

kristip17 commented:
Ready for NLS review.

kristip17 commented:
@jantley-ibm  -- I don't know if you want to give this another eyeball. The first commit was missing the NLS changes and I made some tweaks from the last update.

jvanhill commented on dev/com.ibm.websphere.security.wim.base/src/com/ibm/websphere/security/wim/ConfigConstants.java:
```diff
@@ -716,4 +716,25 @@
+    String CONFIG_PROP_KRB5_AUTH = "krb5Authentication";
+    String CONFIG_PROP_KRB5_PRINCIPAL = "krb5Principal";
+    String CONFIG_PROP_KRB5_TICKET_CACHE = "krb5TicketCache";
+    String CONFIG_PROP_KRB5_CONFIG = "configFile"; // must match the config on kerberos element
```
Not sure we need CONFIG_PROP_KRB5_TICKET_CACHE or CONFIG_PROP_KRB5_CONFIG as that comes in via the kerberos configuration element that already exists. I assume it is a service we can use as a reference and directly access those values.

kristip17 replied:
I was debating keeping or leaving too. I removed, we can always add them back if we need them for something.

jvanhill reviewed (changes requested):
Just one finding for now.

kristip17 commented:
#libbybot

Build failures are known issues.

utle commented on dev/com.ibm.ws.security.wim.adapter.ldap/resources/OSGI-INF/l10n/metatype.properties:
```diff
@@ -451,3 +451,16 @@ contextPoolConfig=Context Pool Properties
+bindAuthMechanism=Bind Authentication Mechanism
+bindAuthMechanism.desc=The authentication mechanism for binding to the LDAP server when searching or modifying an LDAP entry.
+
+bindAuthMechanism.none=Anonymous bind to the directory service.
```
none - don't need anything? o

kristip17 replied:
Do you mean I should say that using `none` doesn't require setting additional attributes? Or we don't need a none option?

kristip17 replied:
I swapped to `bindAuthMechanism.none=Anonymous bind to the directory service. No additional login attributes are required.`

utle commented on dev/com.ibm.ws.security.wim.adapter.ldap/resources/OSGI-INF/l10n/metatype.properties:
```diff
@@ -451,3 +451,16 @@ contextPoolConfig=Context Pool Properties
+krb5Principal=Kerberos Principal Name
+krb5Principal.desc=Kerberos principal name or Kerberos service principal name that is used to authenticate with the Key Distribution Center (KDC) when using the GSSAPI bindAuthMechanism.
+
+krb5TicketCache=Kerberos Ticket Cache File
```
We have these defined in data source, should we keep them the same?
krb5Principal=Kerberos principal name
krb5Principal.desc=The name of the Kerberos principal name or Kerberos service name to be used.

krb5TicketCache=Kerberos ticket cache
krb5TicketCache.desc=The file location where Kerberos credentials for the Kerberos principal name or service name will be stored. Also known as the Kerberos credential cache (ccache)

kristip17 replied:
I did use the krb5TichetCache desc, but I changed the tense. The krb5Princpal is from tWAS and I added `when using the GSSAPI bindAuthMechanism` onto it. Do you want to swap it back to the simpler version from datasource?

utle replied:
I would keep it the same with data source for Liberty.

kristip17 replied:
Will swap to match datasource

utle commented on dev/com.ibm.ws.security.wim.adapter.ldap/resources/OSGI-INF/metatype/metatype.xml:
```diff
@@ -170,6 +170,15 @@
 		<AD id="timestampFormat" name="%timestampFormat" description="%timestampFormat.desc" required="false" type="String" />
 		<!-- End VMM ported Entries -->	
 
+	<AD id="bindAuthMechanism" name="internal" description="internal use only" required="false" type="String">
```
Just wonder why we do not use beta instead of internal? also, should we have default value "simple"

kristip17 replied:
I was going to do initial commit as internal and then when we had useable runtime code, promote to beta.

kristip17 replied:
I was also planning to wait on setting the default to simple until we GA. No default for internal/beta.

kristip17 replied:
Will update to beta when we've delivered the runtime code, will set the default when we GA.

utle commented on dev/com.ibm.ws.security.wim.adapter.ldap/resources/OSGI-INF/metatype/metatype.xml:
```diff
@@ -170,6 +170,15 @@
+	</AD>
+
+	<AD id="krb5Principal" name="internal" description="internal use only" type="String" required="false" />
+	<AD id="krb5TicketCache" name="internal" description="internal use only" type="String" required="false" />
```
We should use beta instead of internal for all the LDAP support Kerberos new metatype

kristip17 replied:
Will set to beta when we deliver the runtime code.

kristip17 commented:
I added an issue for tasks to do when we GA (update the metatype, remove beta fencing) -- #15722

I also added that the runtime issue should change from internal to beta. #15640

kristip17 commented:
#libbybot
